### PR TITLE
Add EntityTag.puffstate property

### PR DESCRIPTION
**Pull request by Hydroxycobalamin**
This PR adds the `EntityTag.puffstate` property for Pufferfishs. 

Tags:

- `<EntityTag.puffstate>` returns the puff state of a Pufferfish.

Mechanism:

- `puffstate` sets the puff state of a Pufferfish.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java [hunk 1]
```diff
@@ -109,6 +109,7 @@ public static void registerMainProperties() {
         PropertyParser.registerProperty(EntityPumpkinHead.class, EntityTag.class);
         PropertyParser.registerProperty(EntityRiptide.class, EntityTag.class);
         PropertyParser.registerProperty(EntityRotation.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityPuffState.class, EntityTag.class);
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: Out of alphabetical order

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.PufferFish;
+
+public class EntityPuffState implements Property {
+
+    public static boolean describes(ObjectTag entity) {
+        if (!(entity instanceof EntityTag)) {
+            return false;
+        }
+        return ((EntityTag) entity).getBukkitEntityType() == EntityType.PUFFERFISH;
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java]: Usually entity properties use `instanceof` checks

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    private EntityPuffState(EntityTag entity) {
+        this.entity = entity;
+    }
+
+    EntityTag entity;
+
+    @Override
+    public String getPropertyString() {
+        return new ElementTag(getPufferFish().getPuffState()).identify();
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java]: `String#valueOf`

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    public static void registerTags() {
+
+        // <---[tag]
+        // @attribute <EntityTag.puffstate>
+        // @returns ElementTag(Number)
+        // @mechanism EntityTag.puffstate
+        // @group properties
+        // @description
+        // Returns the puff state of a Pufferfish.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java]: Should probably include more information - is it `0` to `1`? `0` to `100`?

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        PropertyParser.registerMechanism(EntityPuffState.class, ElementTag.class, "puffstate", (object, mechanism, input) -> {
+            int puffState = input.asInt();
+            object.getPufferFish().setPuffState(puffState);
+        });
+    }
+
+    public PufferFish getPufferFish() {
+        return (PufferFish) entity.getBukkitEntity();
+    }
+}
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java]: Missing NL@EOF

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        // @name puffstate
+        // @input ElementTag(Number)
+        // @description
+        // Sets the puff state of a Pufferfish.
+        // If this is higher than 2 the Pufferfish keeps inflated.
+        // @tags
+        // <EntityTag.puffstate>
+        // -->
+        PropertyParser.registerMechanism(EntityPuffState.class, ElementTag.class, "puffstate", (object, mechanism, input) -> {
+            int puffState = input.asInt();
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityPuffState.java]: Redundant variable, can use `input.asInt()` directly

**Comment by tal5** [comment]
Should be merged into the `EntityTag.size` property

**Comment by Hydroxycobalamin** [comment]
Merged it.

**Pull request merged**